Wiley Online Library.js

```diff
@@ -396,6 +398,21 @@ function addFreeAccessTag(doc, item) {
 	};
 }
 
+function getORCID(doc, item) {
+	let authorOrcidEntries = doc.querySelectorAll('#sb-1 span');
+	for (let authorOrcidEntry of authorOrcidEntries) {
+		let authorEntry = authorOrcidEntry.querySelector('.author-name accordion-tabbed__control, span');
+		let orcidEntry = authorOrcidEntry.querySelector('*[href^="https://orcid"]');
+		if (authorEntry && orcidEntry) {
+			let author = authorEntry.textContent;
+			let orcid = orcidEntry.text.match(/\d+-\d+-\d+-\d+x?/i)[0];
```

Comment:
`[0]` only exists if the regex really matches.

Reply:
yes, that's how it should be.

Reply:
If the match fails you get a null-value, so accessing `[0]` leads to an exception and the translator aborts. So the case that the match fails must be handled separately, e.g. by using 
```javascript
   let orcidMatch  = orcidEntry.text.match(/\d+-\d+-\d+-\d+x?/i);
   let orcid = orcidMatch ? orcidMatch[0] : null;
```
and the handle it appropriately accoring to the value of orcid.
Besides, as the ORCID ID should only be set if it exists, alternatively one could use a construct like
```javascript
if (authorEntry && orcidEntry && orcidEntry.text && orcidEntry.text.match(/\d+-\d+-\d+-\d+x?/i)) {
   let author = authorEntry.textContent;
   let orcid = orcidEntry.text.match(/\d+-\d+-\d+-\d+x?/i)[0];
   ...
}

Reply:
I will use your second suggestion. thanks!